plugins/modules/notification/telegram.py

```diff
@@ -11,44 +11,76 @@
 DOCUMENTATION = '''
 
 module: telegram
-author: "Artem Feofanov (@tyouxa)"
+author:
+ - "Artem Feofanov (@tyouxa)"
+ - "Nikolai Lomov (@lomserman)"
 
 short_description: module for sending notifications via telegram
 
 description:
-    - Send notifications via telegram bot, to a verified group or user
+    - Send notifications via telegram bot, to a verified group or user.
+    - Also, you may try to use any other telegram bot api method, if you specify "api_method" argument.
+      This isn't tested though.
 notes:
     - You will require a telegram account and create telegram bot to use this module.
+    - Deprecated arguments left for backward compatibility, when used - will replace corresponding
+      variables in api_args
 options:
-  msg:
+  token:
     type: str
     description:
-      - What message you wish to send.
+      - Token identifying your telegram bot.
     required: true
+  api_method:
+    type: str
+    description: BOt API method. For reference, see https://core.telegram.org/bots/api
+    default: SendMessage
+  api_args:
+    type: dict
+    description:
+      - Any parameters for method. I.e., for SendMessage, see https://core.telegram.org/bots/api#sendmessage
+  msg:
+    type: str
+    description:
+      - (Deprecated) What message you wish to send.
```

Comment:
@felixfontein Could you please comment on the deprecation cycle guidelines?

Reply:
Resolved somewhere below.